app/src/main/java/de/xikolo/models/Video.kt

```diff
@@ -0,0 +1,131 @@
+package de.xikolo.models
+
+import com.squareup.moshi.Json
+import de.xikolo.models.base.RealmAdapter
+import de.xikolo.models.dao.ItemDao.Unmanaged.Companion.findForContent
+import de.xikolo.utils.LanguageUtil
+import io.realm.RealmList
+import io.realm.RealmObject
+import io.realm.annotations.Ignore
+import io.realm.annotations.PrimaryKey
+import moe.banana.jsonapi2.JsonApi
+import moe.banana.jsonapi2.Resource
+
+open class Video : RealmObject() {
+
+    @PrimaryKey
+    var id: String? = null
+
+    var summary: String? = null
+
+    var duration: Int = 0
+
+    var singleStream: VideoStream? = null
+
+    var lecturerStream: VideoStream? = null
+
+    var slidesStream: VideoStream? = null
+
+    var slidesUrl: String? = null
+
+    var slidesSize: Int = 0
+
+    var audioUrl: String? = null
+
+    var audioSize: Int = 0
+
+    var transcriptUrl: String? = null
+
+    var transcriptSize: Int = 0
+
+    var thumbnailUrl: String? = null
+
+    var subtitles = RealmList<VideoSubtitles>()
+
+    @Ignore
+    var item: Item? = findForContent(id)
+
+    // local field
+    var progress = 0
+
+    val streamToPlay: VideoStream
+        get() {
+            return if (singleStream != null && (singleStream!!.hdUrl != null || singleStream!!.sdUrl != null)) {
```

Comment:
The `singleStream!!` call is still unsafe. Consider using `singleStream?.hdUrl != null`.